plugins/flatpak/gs-flatpak.c

```diff
@@ -2755,6 +2756,19 @@ gs_flatpak_refine_appstream_from_bytes (GsFlatpak *self,
 
 	fixup_flatpak_appstream_xml (source, origin);
 
+	/* add metadata */
+	if (installed_ref != NULL) {
+		g_autoptr(XbBuilderNode) info = NULL;
+		g_autofree char *icon_prefix = NULL;
+
+		info = xb_builder_node_insert (NULL, "info", NULL);
+		xb_builder_node_insert_text (info, "scope", as_app_scope_to_string (self->scope), NULL);
+		icon_prefix = g_build_filename (flatpak_installed_ref_get_deploy_dir (installed_ref),
+						"files", "share", "app-info", "icons", "flatpak", NULL);
+		xb_builder_node_insert_text (info, "icon-prefix", icon_prefix, NULL);
+		xb_builder_source_set_info (source, info);
```

Comment:
Is it possible that the appstream data already has icon set?

Reply:
Yes, the appstream data will have an `<icon>` but not an icon prefix, which is machine specific.